From bbf0c5a6515fa1b2661d104557365a2e25f1554f Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Thu, 22 Feb 2018 18:42:55 +0200
Subject: [PATCH 110/706] enetc: check that PF driver doesn't probe a VF

if it does probe the VF it crashes.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
(cherry picked from commit f7d3dd1d0d9925868769b2b6563c8f600077e190)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_pf.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_pf.c b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
index 316eb9f6b930..cad80cc87499 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_pf.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
@@ -494,6 +494,11 @@ static int enetc_pf_probe(struct pci_dev *pdev,
 	}
 
 	si = pci_get_drvdata(pdev);
+	if (!si->hw.port || !si->hw.global) {
+		err = -ENODEV;
+		dev_err(&pdev->dev, "could not map PF space, probing a VF?\n");
+		goto err_map_pf_space;
+	}
 
 	enetc_get_si_caps(si);
 
@@ -553,6 +558,7 @@ static int enetc_pf_probe(struct pci_dev *pdev,
 	si->ndev = NULL;
 	free_netdev(ndev);
 err_alloc_netdev:
+err_map_pf_space:
 	enetc_pci_remove(pdev);
 
 	return err;
-- 
2.17.1

